### PR TITLE
Refactor: date-fns migration

### PR DESCRIPTION
**Pull request by luixo**
Comparison of bundles:

Before:
```
static/chunks/pages/_app-2015e57b472bada4.js
./node_modules/moment (77.26 KB)

Total: 77.26KB
```

After:
```
static/chunks/274-9c3faef4dafd5241.js
./node_modules/date-fns/esm (2.95 KB)

static/chunks/109-b5cb8dc3b25400ee.js
./node_modules/date-fns/esm (4.03 KB)

static/chunks/87-7ac469896b773dd4.js
./node_modules/date-fns/esm (1.47 KB)

static/chunks/pages/blocks-559ab4cf7173d7a3.js
./node_modules/date-fns/esm (1.47 KB)

static/chunks/282-cee615f2e67b7d95.js // format fn
./node_modules/date-fns/esm (20.64 KB)

static/chunks/933.b0f72a2fd31c6f7d.js // en locale
./node_modules/date-fns/esm/locale (7.33 KB)

static/chunks/351.1ea77d5d22b6c9af.js // ru locale
./node_modules/date-fns/esm (3.2 KB)

static/chunks/473.efad477406d7a5ba.js // uk locale
./node_modules/date-fns/esm (3.2 KB)

static/chunks/215.3ff08bd7b69e16ee.js // vi locale
./node_modules/date-fns/esm/locale/_lib (1.7 KB)

static/chunks/489.5a4293284963da18.js // zh-Hans locale
./node_modules/date-fns/esm (3.2 KB)

Total: depending on the page, from ~20KB to ~30KB
```

Also, I made locale loadable async so we user fetch locales he doesn't use.

### CODE REVIEW

File: frontend/src/libraries/date-locale.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      return (await import("./locales/en")).locale;
+    case "ru":
+      return (await import("./locales/ru")).locale;
+    case "vi":
+      return (await import("./locales/vi")).locale;
+    case "zh-Hans":
+      return (await import("./locales/zh-Hans")).locale;
+    case "uk":
+      return (await import("./locales/uk")).locale;
+    case "cimode":
```
Comment (shelegdmitriy) [inline frontend/src/libraries/date-locale.ts:21]: This case is needed for test snapshots, if so maybe we should add some explaining comment here?

File: frontend/src/libraries/locales/uk.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    few: "секунди",
+    many: "секунд",
+  },
+  minutes: {
+    one: "хвилину",
+    few: "хвилини",
+    many: "хвилин",
+  },
+  hours: {
+    one: "гадина",
```
Comment (shelegdmitriy) [inline frontend/src/libraries/locales/uk.ts:17]:
```suggestion
    one: "година",
```

**Review by shelegdmitriy — CHANGES_REQUESTED** [review]
Looks good to me

File: frontend/src/components/nodes/NodesEpoch.tsx [hunk 3]
```diff
@@ -51,9 +52,11 @@ const NodesEpoch: React.FC<Props> = React.memo(
   ({ epochStartHeight, epochLength, epochStartTimestamp }) => {
     const { t } = useTranslation();
     const latestBlockSub = useSubscription(["latestBlock"]);
+    const format = useDateFormat();
     if (latestBlockSub.status !== "success") {
       return null;
     }
+    console.log("late", latestBlockSub.data, epochStartHeight, epochLength);
```
Comment (shelegdmitriy) [inline frontend/src/components/nodes/NodesEpoch.tsx:59]: ?

**Review by shelegdmitriy — CHANGES_REQUESTED** [review]
Please address comments and we are ready to merge it I think

File: frontend/src/components/blocks/__tests__/__snapshots__/BlocksRow.test.tsx.snap [hunk 4]
```diff
@@ -88,7 +88,7 @@ exports[`<BlocksRow /> renders 1`] = `
             </span>
               
             <span>
-              1s ago
+              0s
```
[the remark this answers is not in the document]
Comment (shelegdmitriy) [reply frontend/src/components/blocks/__tests__/__snapshots__/BlocksRow.test.tsx.snap:91]: @luixo, sorry I missed that this is only for `cimode` 🤦‍♂️

**Comment by luixo** [comment]
> Bundle size diff

Okay that's weird 🤔

**Comment by frol** [comment]
> Okay that's weird 🤔

Yeah, lol, definitely not the outcome we wanted to see :-D

**Comment by luixo** [comment]
> Yeah, lol, definitely not the outcome we wanted to see :-D

Okay, manual analysis helped me out with that.

The "app" chunks gets:
* -170Kb of moment.js
* +18Kb of default english
* +18Kb of common date-fns functions

Beside that we have in separate chunks:
* +20Kb of vi locale
* +29Kb of ru locale
* +28Kb of uk locale
* +18Kb of en locale
* +18Kb of zh-Hans locale
* +18Kb of default english + 67Kb of formatting

That's getting as 64Kb of delta in chunks, but in actual loads we get less Kbs.
Not much less, really, as we have two default english locales loaded separately and some common function also intersect twice.
I think I'll get back into more granular chunks splitting later.

**Pull request merged**